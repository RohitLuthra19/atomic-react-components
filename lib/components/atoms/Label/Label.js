// @flow
import React from 'react';
import type { Node } from 'react';

import { getLabelFromChildren } from '../../../core/utils/elementLocator';

type Props = {
  className?: string,
  elementLocator?: string,
  htmlFor: string,
  children: any,
};

const Label = ({ className, htmlFor, children, elementLocator, ...others }: Props): Node => (
  /* eslint-disable jsx-a11y/label-has-for */
  // FIXME: Remove the eslint comment above
  <label
    className={className}
    htmlFor={htmlFor}
    data-locator={elementLocator || `label-${htmlFor}-${getLabelFromChildren(children)}`}
    {...others}
  >
    {children}
  </label>
);

Label.defaultProps = {
  className: '',
  elementLocator: '',
};

export default Label;
