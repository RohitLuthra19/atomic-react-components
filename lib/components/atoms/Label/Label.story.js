// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

// Import Vanilla Component to display markup and props
import Label from './Label';
import Input from '../Input';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module).addWithChapters('Label', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <React.Fragment>
              <Label htmlFor="userid"> User Name </Label>
              <Input id="userid" name="userid" />
            </React.Fragment>
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
  ],
});
