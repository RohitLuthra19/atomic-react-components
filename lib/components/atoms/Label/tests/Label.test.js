import React from 'react';
import { shallow } from 'enzyme';
import Label from '../Label';

describe('Label Component', () => {
  let label = '';

  test('should render correctly', () => {
    label = shallow(<Label htmlFor="userid"> User Name </Label>);
    expect(label).toMatchSnapshot();
  });
});
