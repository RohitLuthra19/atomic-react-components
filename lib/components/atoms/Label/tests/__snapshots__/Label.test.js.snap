// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render correctly 1`] = `
<label
  className=""
  htmlFor="userid"
>
   User Name 
</label>
`;
